applications/StructuralMechanicsApplication/custom_constitutive/hyper_elastic_isotropic_kirchhoff_3d.cpp

```diff
@@ -165,79 +165,55 @@ void HyperElasticIsotropicKirchhoff3D::CalculateMaterialResponseCauchy (Constitu
 
 void HyperElasticIsotropicKirchhoff3D::InitializeMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
 {
-//     rValues.Set(ConstitutiveLaw::INITIALIZE_MATERIAL_RESPONSE);

```

Comment:
maybe add a warning saying that the element should check if this function needs to be called?

BTW what abt the linear elastic laws? Isn't it the same there?

Reply:
Yes, it is already there

Reply:
I will not add any warning, a lot of verbosity...

Reply:
but then it will never be noticed/updated :/
if you use `KRATOS_WARNING_ONCE` you will get the warning only once in the entire run, I think this would be appropriate

Reply:
I think I have a better idea, add a check for the error in the base class. The error is only thrown when the compute is mandatory (madnatory by default). this will reduce a lot of empty code

Reply:
@philbucher what do you think about my proposal?, now it is errored only when it is required

Reply:
it is fine but without a warning we will never know which elements do not implement the correct version so we will waste a lot of unnecessary computations

Reply:
Well, I understand your concern, but the default is an error, all CL without overriding the method will crash (as it does right now)

Reply:
but the baseclass says that the Initialize & finalize is required by default, unless you override it

Reply:
in the base class you should NOT throw an error. IT is allowed to call it, just slower